src/query/functions/temporal/aggregation.go

```diff
@@ -134,16 +136,18 @@ func makeHoltWintersFn(sf, tf float64) aggFunc {
 			b                       float64
 			x, y                    float64
 			s0, s1                  float64
+			idx                     int
 		)
 
-		for i, val := range vals {
+		for _, val := range vals {
 			if math.IsNaN(val) {
 				continue
 			}
 
 			if !foundFirst {
 				foundFirst = true
 				s1 = val
+				idx++
```

Comment:
Shouldn't this be outside of the `if !foundFirst` check? Otherwise, isn't it lagging by 1 behind the actual non-nan index?

Reply:
Yeah I added `idx++` at the end to avoid this, but I can just add to right before here.